lib/NormalModuleFactory.js

```diff
@@ -227,23 +230,63 @@ class NormalModuleFactory extends ModuleFactory {
 			}
 		);
 		this.hooks.resolve.tapAsync(
-			/** @type {TODO} */ ({
+			{
 				name: "NormalModuleFactory",
 				stage: 100
-			}),
+			},
 			(data, callback) => {
+				let { resolveOptions } = data;
 				const {
 					contextInfo,
 					context,
+					dependencies,
 					request,
-					resolveOptions,
 					fileDependencies,
 					missingDependencies,
 					contextDependencies
 				} = data;
-
+				const category =
+					dependencies.length > 0 ? dependencies[0].category : undefined;
 				const loaderResolver = this.getResolver("loader");
-				const normalResolver = this.getResolver("normal", resolveOptions);
+
+				if (category === Dependency.Categories.CommonJS) {
+					if (!resolveOptions) {
+						resolveOptions = REQUIRE_CONDITION_NAME;
+					} else {
+						if (Array.isArray(resolveOptions.conditionNames)) {
+							if (resolveOptions.conditionNames.indexOf("require") === -1) {
+								resolveOptions = { ...resolveOptions };
+								resolveOptions.conditionNames = REQUIRE_CONDITION_NAME.conditionNames.slice();
+							}
+						} else {
+							resolveOptions = {
```

Comment:
You can't modify the resolveOptions like that. That would break all the caching..